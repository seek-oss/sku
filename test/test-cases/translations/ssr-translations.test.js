/* eslint-disable jest/expect-expect */
const path = require('path');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const { getAppSnapshot } = require('../../utils/appSnapshot');
const waitForUrls = require('../../utils/waitForUrls');
const skuStartConfig = require('./app/sku-ssr.config');

const appDir = path.resolve(__dirname, 'app');

const getTestConfig = (skuConfig) => ({
  backendUrl: `http://localhost:${skuConfig.serverPort}`,
  targetDirectory: path.join(__dirname, skuConfig.target),
});

describe('ssr translations', () => {
  let server;
  const { backendUrl } = getTestConfig(skuStartConfig);

  beforeAll(async () => {
    server = await runSkuScriptInDir('start-ssr', appDir, [
      '--config=sku-ssr.config.js',
    ]);
    await waitForUrls(backendUrl);
  });

  afterAll(async () => {
    await server.kill();
  });

  it('should render en', async () => {
    const app = await getAppSnapshot(`${backendUrl}/en`);
    expect(app).toMatchSnapshot();
  });

  it('should render fr', async () => {
    const app = await getAppSnapshot(`${backendUrl}/fr`);
    expect(app).toMatchSnapshot();
  });

  it('should render en-PSEUDO', async () => {
    const app = await getAppSnapshot(`${backendUrl}/en?pseudo=true`);
    expect(app).toMatchSnapshot();
  });
});
