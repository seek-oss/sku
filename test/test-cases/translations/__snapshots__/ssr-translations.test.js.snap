// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssr translations should render en 1`] = `
SCRIPTS: Array [
  "http://127.0.0.1:8313/runtime.js",
  "http://127.0.0.1:8313/vendors-node_modules_loadable_component_dist_loadable_esm_js-node_modules_pmmmwh_react-refres-ac261c.js",
  "http://127.0.0.1:8313/main.js",
  "http://127.0.0.1:8313/en-translations.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="en-translations"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      ["en-translations"]
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
            crossorigin="anonymous"
    >
      {"namedChunks":["en-translations"]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="en-translations"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`ssr translations should render en-PSEUDO 1`] = `
SCRIPTS: Array [
  "http://127.0.0.1:8313/runtime.js",
  "http://127.0.0.1:8313/vendors-node_modules_loadable_component_dist_loadable_esm_js-node_modules_pmmmwh_react-refres-ac261c.js",
  "http://127.0.0.1:8313/main.js",
  "http://127.0.0.1:8313/en-PSEUDO-translations.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="en-PSEUDO-translations"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        [Ḩẽẽẽƚƚööö šķǚǚǚ]
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      ["en-PSEUDO-translations"]
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
            crossorigin="anonymous"
    >
      {"namedChunks":["en-PSEUDO-translations"]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="en-PSEUDO-translations"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`ssr translations should render fr 1`] = `
SCRIPTS: Array [
  "http://127.0.0.1:8313/runtime.js",
  "http://127.0.0.1:8313/vendors-node_modules_loadable_component_dist_loadable_esm_js-node_modules_pmmmwh_react-refres-ac261c.js",
  "http://127.0.0.1:8313/main.js",
  "http://127.0.0.1:8313/fr-translations.js",
]
CSS: Array []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[0]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[1]"
          crossorigin="anonymous"
    >
    <link data-chunk="main"
          rel="preload"
          as="script"
          href="scripts[2]"
          crossorigin="anonymous"
    >
    <link data-chunk="fr-translations"
          rel="preload"
          as="script"
          href="scripts[3]"
          crossorigin="anonymous"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Bonjour sku
      </div>
    </div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__"
            type="application/json"
            crossorigin="anonymous"
    >
      ["fr-translations"]
    </script>
    <script id="__LOADABLE_REQUIRED_CHUNKS___ext"
            type="application/json"
            crossorigin="anonymous"
    >
      {"namedChunks":["fr-translations"]}
    </script>
    <script async
            data-chunk="main"
            src="scripts[0]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[1]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="main"
            src="scripts[2]"
            crossorigin="anonymous"
    >
    </script>
    <script async
            data-chunk="fr-translations"
            src="scripts[3]"
            crossorigin="anonymous"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
